core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -774,6 +774,19 @@ protected void convertSelectImpl(
         cluster.traitSet().canonize(RelCollations.of(collationList));
 
 
+    /** The behavior of QUALIFY is the same as if you were to add the window function condition
```

Comment:
Originally, this was present in both convertSelectList and convertAgg. I pulled it up because I thought it made the code path cleaner.